### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by aran159**
PR for fixing the errors found when building with the new changes merged into master

### CODE REVIEW

**Comment by aran159** [comment]
Eigen3 was already being installed in `install.sh`. See line 114:

`sudo apt install build-essential cmake git libpoco-dev libeigen3-dev`

**Comment by aran159** [comment]
@krishnachaitanya9 @watakandai Do you know why setting up Eigen3 as a ros package worked in Caleb's and my computers and not in others'?

**Comment by watakandai** [comment]
> @krishnachaitanya9 @watakandai Do you know why setting up Eigen3 as a ros package worked in Caleb's and my computers and not in others'?

I'm not really good at C++, but my guess is that when Caleb built (installed) ROS Melodic, it was able to find `Eigen` and may have linked to `catkin`, but yours didn't somehow. 
That's my best guess.

@krishnachaitanya9 What do you think?

**Comment by krishnachaitanya7** [comment]
Hi all! Because you had it in your src/ folder in your ROS project. But others hadn't. If they download Eigen and put it in src/ folder, your old code should work for them

File: CMakeLists.txt [hunk 1]
```diff
@@ -121,6 +124,7 @@ catkin_package(CATKIN_DEPENDS xacro message_runtime)
 include_directories(
   include
   ${catkin_INCLUDE_DIRS}
+  ${Eigen3_INCLUDE_DIRS}
```
Comment (krishnachaitanya7) [inline CMakeLists.txt]: I think there is no need for this. I was able to run without this yesterday night.

File: CMakeLists.txt [hunk 2]
```diff
@@ -157,9 +162,9 @@ add_dependencies(CartesianPositionController ros_robotic_skin_gencpp)
 
 ## Specify libraries to link a library or executable target against
 target_link_libraries(KDL_server ${catkin_LIBRARIES})
-target_link_libraries(proximity_listener ${catkin_LIBRARIES})
-target_link_libraries(proximity_visualizer ${catkin_LIBRARIES})
-target_link_libraries(CartesianPositionController ${catkin_LIBRARIES})
+target_link_libraries(proximity_listener ${catkin_LIBRARIES} ${Eigen3_LIBRARIES})
+target_link_libraries(proximity_visualizer ${catkin_LIBRARIES} ${Eigen3_LIBRARIES})
+target_link_libraries(CartesianPositionController ${catkin_LIBRARIES} ${Eigen3_LIBRARIES} CGAL::CGAL CGAL::CGAL_Core)
```
Comment (krishnachaitanya7) [inline CMakeLists.txt]: Eigen3 is header only: https://packages.ubuntu.com/bionic/all/libeigen3-dev/filelist. There is no need for linking. But I can be wrong too. Lemme know!

**Review by krishnachaitanya7 — CHANGES_REQUESTED** [review]

**Comment by watakandai** [comment]
> Hi all! Because you had it in your src/ folder in your ROS project. But others hadn't. If they download Eigen and put it in src/ folder, your old code should work for them

But you shouldn't add Eigen to your `src/` folder, but include from INCLUDE_DIR. 
This is just for testing right?

**Comment by krishnachaitanya7** [comment]
Yeah, Eigen shouldn't be added like that. Not a best practice. It will just make the build work.

**Comment by aran159** [comment]
Yes! We never added Eigen in the src directory.
With the changes working for Kedar. Can you @krishnachaitanya9 @peasant98 @watakandai check that it builds?
@krishnachaitanya9 I think that I was adding eigen before as a ros package doing `apt install ros-melodic-eigen-stl-containers`. That could be the reason why it wasn't working for anyone else but Caleb and I, but I am not sure.
Now we're doing it as we talked yesterday.
Thank you for reviewing!

**Comment by watakandai** [comment]
I'll run the script today and see what happens.

**Review by krishnachaitanya7 — APPROVED** [review]
LGTM if Flake8 i sokay

File: install.sh [hunk 3]
```diff
@@ -109,9 +109,9 @@ sudo apt install ros-melodic-imu-filter-madgwick ros-melodic-ros-control ros-mel
 cd src
 git clone git@github.com:HIRO-group/Custom_IMU_Madgwick_Filter.git
 # if franka build is desired from source
+sudo apt install build-essential cmake git libpoco-dev libeigen3-dev
```
Comment (peasant98) [inline install.sh:112]: nice

File: src/JointVelocityController.cpp [hunk 4]
```diff
@@ -37,8 +37,10 @@ void JointVelocityController::sendVelocities(const Eigen::VectorXd vel)
 {
     if (vel.size() == 7)
     {
-        for (int i = 0; i < 7; i++)
-            publishers[i].publish(vel[i]);
+        for (int i = 0; i < 7; i++){
```
Comment (peasant98) [inline src/JointVelocityController.cpp:40]: nit: I would have same bracket placement for if statements as well as for loop statements - maybe have this one on a new line then?

**Pull request merged**